File: C++/gcd.cpp
 // first approach
 #include <iostream>

 using namespace std;

 long long gcd(long long a, long long b)
 {
     return (b == 0LL ? a : gcd(b, a % b));
 }

 int main()
 {
     long long a, b;
     cin >> a >> b;
     cout << gcd(a, b) << '\n';
     return 0;
 }

 // second approach
#include <iostream>
using namespace std;

long long gcd(long long x, long long y){
    if(y == 0LL) return x;
    return gcd(y, x % y);
}
int main() {
    int y; int x;
    cin >> y >> x;
    cout << gcd(x, y) << '\n';
    return 0;
}

File: Python/gcd.py
import sys
sys.setrecursionlimit(10000000)


### first approach
x, y = list(map(int, input().split()))

def recursion(x, y):
    if y == 0:
        return x

    return recursion(y, x%y)
print(recursion(x, y))

### second approach
def gcd(a, b):
    return a if b == 0 else gcd(b, a % b)


a, b = [int(x) for x in input().split()]
print(gcd(a, b))

### third approach
x, y = list(map(int, input().split()))

while True:
    if y != 0 and x != 0:
        if y > x:
            z = y%x
            y = z
        else:
            z = x%y
            x = y
            y = z
    else:
        [print(x) if y == 0 else print(y)]
        break


### forth approach
a, b = [int(x) for x in input().split()]

while b != 0:
    a, b = b, a % b
print(a)